docs/int/openapi.yaml

```diff
@@ -796,6 +796,196 @@ paths:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JobCode'
+    delete:
+      summary: Remove a Job Code
+      description: Delete a Job Code from OSMT
+      tags:
+        - MetaData
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TaskResult'
+    post:
+      summary: Update a Job Code
+      description: Update an existing Job Code
+      tags:
+        - MetaData
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/JobCode'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JobCode'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+  /api/metadata/named-references:
+    get:
+      summary: Get all Named References of a type
+      description: Returns all Named References of a specific Type
+      tags:
+        - MetaData
+      parameters:
+        - $ref: '#/components/parameters/NameReferenceType'
+        - $ref: '#/components/parameters/PaginationSizeQueryRequired'
+        - $ref: '#/components/parameters/PaginationFromQueryRequired'
+        - $ref: '#/components/parameters/SortOrderQuery'
+      responses:
+        '200':
+          description: OK
+          headers:
+            X-Total-Count:
+              $ref: '#/components/headers/XTotalCount'
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/NamedReferenceResponse'
+    post:
+      summary: Define a Named Reference
+      description: Create a new Named Reference in OSMT
+      tags:
+        - MetaData
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/NamedReferenceRequest'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/NamedReferenceResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+
+  /api/metadata/named-references/{id}:
+    parameters:
+      - $ref: '#/components/parameters/IdPath'
+    get:
+      summary: Retrieve a Named Reference
+      description: Retrieve all the information about a Named Reference
+      tags:
+        - MetaData
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NamedReferenceResponse'
+    delete:
```

Comment:
- [ ] Same request for /remove & /update for these delete & update endpoint as jobcodes